src/comment.cpp

```diff
@@ -304,7 +304,7 @@ void CommentParser::collectParameters(const char *fulltext, FileModule *root_mod
 
 		// Look for the group to which the given assignment belong
 		int i=0;
-		for (;i<groupList.size() && groupList[i].lineNo<firstLine;i++);
+		for (; (unsigned)i < groupList.size() && groupList[i].lineNo<firstLine;i++);
```

Comment:
The original code finds the last group with lineNo less than firstLine. Marius' version finds the first group with lineNo >= firstLine, which is out by one. Your suggestion would find the first group with lineNo < firstLine.

To do it Marius style I think the loop needs to iterate backwards but I don't know how you do that with the C++00x version of for.